src/main/c/openssl/extconf.rb

```diff
@@ -30,11 +30,10 @@
 
 if ENV['OPENSSL_PREFIX']
   $CFLAGS += " -I #{ENV['OPENSSL_PREFIX']}/include"
-  $LIBS += " -l #{ENV['OPENSSL_PREFIX']}/lib/libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
-else
-  $LIBS += " -l libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
 end
 
+$LIBS += " -l libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
```

Comment:
e.g.: on Darwin, when installing a gem that directly links against libssl (Puma does for instance), it will by default pick the Homebrew libssl, but use the system openssl.h header, which not going to work.

BTW, installing a gem linking to a native lib also supports this kind of dir option (via `dir_config` in the extconf.rb):
```
gem install mysql2 -- --with-mysql-dir=/usr/local/mysql
```

But indeed, this is build-time config and here we are rather talking about link-time config, which usually doesn't need headers. It might be good to unify them though, for simplicity.